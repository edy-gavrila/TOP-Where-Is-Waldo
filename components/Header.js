import React from "react";
import Image from "next/image";
function Header({ title, icon, menu }) {
  return (
    <div className="flex items-center w-full bg-purple-700 text-white h-12 px-2 shadow-md shadow-purple-500">
      <div className="flex mr-4">
        <Image src={icon} height="40" width="40" alt="" layout="fixed" />
      </div>
      <h1 className="text-2xl mr-4">{title}</h1>
      <div>{menu}</div>
    </div>
  );
}

export default Header;