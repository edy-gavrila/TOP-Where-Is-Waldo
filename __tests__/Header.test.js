import { render } from "@testing-library/react";
import Header from "../components/Header";
import headerLogo from "../assets/img/header-logo.png";

describe("'Header' component tests:", () => {
  test("It renders correctly:", () => {
    const { container } = render(
      <Header icon={headerLogo} title="Where's that thing?" />
    );
    expect(container).toMatchSnapshot();
  });
});
