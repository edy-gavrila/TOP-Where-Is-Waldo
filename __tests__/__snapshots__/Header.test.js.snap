// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'Header' component tests: It renders correctly: 1`] = `
<div>
  <div
    class="flex items-center w-full bg-purple-700 text-white h-12 px-2 shadow-md shadow-purple-500"
  >
    <div
      class="flex mr-4"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 40px; height: 40px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt=""
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <h1
      class="text-2xl mr-4"
    >
      Where's that thing?
    </h1>
    <div />
  </div>
</div>
`;
